rfcs/0133-ipfs.md

```diff
@@ -0,0 +1,284 @@
+---
+feature: ipfs
+start-date: (fill me in with today's date, YYYY-MM-DD)
+author: John Ericsion (@Ericson2314) on behalf of [Obsidian Systems](https://obsidian.systems)
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Integrate Nix with IPFS, in phases of increasing sophistication.
```

Comment:
While I think most of the proposals in this RFC make sense, I think that proposing them all together makes discussion and acceptance difficult. I would recommend that each proposal is split into its own RFC to keep each change atomic. The whole roadmap is very useful to see however I think this roadmap can be maintained outside of the RFC process or as a bunch of draft RFCs that can then be opened and progressed once the dependencies have been accepted.

Reply:
Well if every step gets its own RFC it will take quite a while!

I am hoping in the first meeting we can chop this into two parts parts, move the second to future work and circle back to it later. Maybe we do that a few times, but I hope the resulting partition ends up being more than one step per RFC :)

Reply:
Yeah, maybe coarse chops are the right approach.

It is important to remember:

1. The time to review an RFC is super-linear with the size.
2. The number of smaller discussions to be resolved is higher for a large RFC.

So while smaller RFCs may have more bureaucratic overhead and organizational overhead it may help actually solidify each component and get it approved. Because trying to understand and discuss the later changes when the assumptions is is based on are mutable provide valuable flexibility but this flexibility also makes it hard to keep discussions concrete and productive.

Reply:
Yeah right it's an optimization problem with overhead on balkanized discussion on one hand vs superlinearity of review cost w.r.t RFC on the other.

Moving things on first meeting into the "future work" section is a sort of non-balanced bisection algorithm to approximate an optimal solution :).

Reply:
I agree, but they also are arguably more "invasive" depending on exactly where we draw the line.

Reply:
Collapsing IPFS threads.